src/common/components/RatingSection/RatingSection.tsx

```diff
@@ -0,0 +1,54 @@
+import {
+  ratingSectionTheme,
+  type RatingSectionVariants,
+} from "./RatingSection.theme";
+import { HeartIcon } from "@/common/components/CustomIcon";
+import { LockCtaOverlay } from "@/common/components/LockCtaOverlay/LockCtaOverlay";
+import { StatItem } from "@/common/components/StatItem";
+import { formatPercentage } from "@/common/functions/format";
+
+export type RatingItem = {
+  label: string;
+  rating: number;
+};
+
+export type RatingSectionProps = RatingSectionVariants & {
+  headingRatingItem: RatingItem;
+  ratingItems: RatingItem[];
+  isLocked?: boolean;
+};
+
+export const RatingSection = ({
+  headingRatingItem,
+  ratingItems,
+  isLocked,
+}: RatingSectionProps) => {
+  const { wrapper, headingContainer, headingRating, statItemWrapper, icon } =
+    ratingSectionTheme();
+  return (
+    <div className={wrapper()}>
+      {isLocked && <LockCtaOverlay />}
+      <div className={headingContainer()}>
+        <div className={headingRating()}>
+          <HeartIcon className={icon()} />
+          <StatItem
+            label={headingRatingItem.label}
+            rating={headingRatingItem.rating}
+            layout="horizontal"
+            isLocked={isLocked}
+          />
+        </div>
+      </div>
+      <div className={statItemWrapper()}>
+        {ratingItems.map((item, i) => (

```

Comment:
Once we use StatItem[] then we can just spread it in, with the same overrides you have e.g isLocked

Reply:
updated at a0c73fe!